File: Algoritmos Matemáticos/FatoracaoComCrivo.java
import java.util.ArrayList;
import java.util.Scanner;

public class FatoracaoComCrivo {
    public static void main(String[] args) {
        //Limite
        int l;
        //Valor
        int v;
        //Encontrados
        ArrayList<Integer> e = new ArrayList<>();
        //Testáveis
        ArrayList<Integer> t = new ArrayList<>();
        //Fatores
        ArrayList<Integer> f = new ArrayList<>();

        Scanner s = new Scanner(System.in);

        System.out.println("Diga o valor v a ser fatorado, tal que v E I; v > 1");
        v = s.nextInt();

        //Gerar os valores testáveis
        for (int i = 2; i <= v; i++) {
            t.add(i);
        }

        //Aplicar o crivo
        while(t.size() > 0) {
            int m = t.get(0);
            e.add(m);
            for (int j = 0; j < t.size(); j++) {
                System.out.println(t.get(j));
                if (t.get(j) % m == 0) {
                    t.remove(j);
                }
            }
        }

        System.out.println(e);

        //Fatorar
        boolean fatorado = false;
        while (!fatorado) {
            if (v == 1) {
                fatorado = true;
            }
            for (int i = 0; e.get(i) <= v; i++) {
                if (v % e.get(i) == 0) {
                    f.add(e.get(i));
                    v /= e.get(i);
                    i--;
                }
                if (e.size() == i + 1) {
                    break;
                }
                fatorado = true;
            }

            //Mostrar resultado
            System.out.println("Os fatores são:");
            for (Integer a : f) {
                System.out.println(a);
            }

        }
    }
}

File: Algoritmos Matemáticos/Teoria de Numeros/FatoracaoComCrivo.py
import math
encontrados = []
testaveis = []
fatores = []

valor = int(input("Diga o valor v a ser fatorado, tal que v E I; v > 1: "))

for i in range(2,math.floor(valor**0.5)):
    testaveis.append(i)

while len(testaveis) > 0:
    m = testaveis[0]
    encontrados.append(m)
    j = 0
    while j < len(testaveis):
        if testaveis[j] % m == 0:
            testaveis.pop(j)
        j+=1

fatorado = False
while not fatorado:
    if valor == 1:
        fatorado = True
    i = 0
    while encontrados[i] <= valor:
        if valor % encontrados[i] == 0:
            fatores.append(encontrados[i])
            valor /= encontrados[i]
            i-=1
        if len(encontrados) == i + 1:
            break
        fatorado = True
        i+=1

print("Os fatores são:")
print(fatores)
